Bump @0x/providers

topliceanu opened the pull request:
### Why?

To unblock this PR https://github.com/smartcontractkit/monitoring/pull/94, in particular this change https://github.com/smartcontractkit/monitoring/pull/94/files#diff-220183e10ab1239286a015ebf08365edL1

#### How?

The branch `feature-173541379/submissionreceived-integration-test` of `monitoring/flux-emulator` will not install on node v12 because scrypt does not compile:
```
warning Error running install script for optional dependency: "/Users/alexandru/work/smartcontract/monitoring/flux_emulator/node_modules/scrypt: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: /Users/alexandru/work/smartcontract/monitoring/flux_emulator/node_modules/scrypt
....
```
`script` is a transient dependency. I tracked it down to @chainlink/test-helpers:
```
➜  flux_emulator git:(feature-173541379/submissionreceived-integration-test) ✗ yarn why scrypt
yarn why v1.22.4
[1/4] 🤔  Why do we have the module "scrypt"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "scrypt@6.0.3"
info Reasons this module exists
   - "@chainlink#test-helpers#@0x#subproviders#ganache-core#ethereumjs-wallet#scrypt.js" depends on it
   - Hoisted from "@chainlink#test-helpers#@0x#subproviders#ganache-core#ethereumjs-wallet#scrypt.js#scrypt"
✨  Done in 0.73s.
```
Updating `@0x/subproviders` in `evm-test-helpers` to the latest version seems to have solved the problem!

HenryNguyen5 commented:
@topliceanu Mind commiting the changed lockfile too? Otherwise installs with —frozen-lockfile will  fail. Also looks like we dont need of the patch files anymore or one of the directory of it needs to be changed.

topliceanu commented:
Thanks @HenryNguyen5! I didn't realise that the build was failing when I opened this one. 
I added the lock file and dropped one of the patches; CI should pass now!
